packages/actor-rdf-parse-html-script/lib/ActorRdfParseHtmlScript.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  ActorRdfParseFixedMediaTypes,
+  IActionRdfParse,
+  IActionRootRdfParse,
+  IActorOutputRootRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+  IActorTestRootRdfParse,
+} from "@comunica/bus-rdf-parse";
+import {ActionContext, Actor, Mediator} from "@comunica/core";
+import { namedNode, quad } from "@rdfjs/data-model";
+import {Readable} from "stream";
+
+/**
+ * A HTML script RDF Parse actor that listens on the 'rdf-parse' bus.
+ *
+ * It is able to extract and parse any RDF serialization from script tags in HTML files
+ * and announce the presence of them by media type.
+ */
+export class ActorRdfParseHtmlScript extends ActorRdfParseFixedMediaTypes {
+
+  public mediatorRdfParse: Mediator<Actor<IActionRootRdfParse, IActorTestRootRdfParse,
+    IActorOutputRootRdfParse>, IActionRootRdfParse, IActorTestRootRdfParse, IActorOutputRootRdfParse>;
+
+  private readonly htmlparser: any;
+
+  constructor(args: IActorRdfParseFixedMediaTypesArgs) {
+    super(args);
+    this.htmlparser = require("htmlparser2");
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext):
+    Promise<IActorRdfParseOutput> {
+    const supportedTypes: string[] = Object.keys((await this.mediatorRdfParse
+      .mediate({
+        context,
+        mediaTypes: true,
+      })).mediaTypes);
+    supportedTypes.push("application/ld+json");
+
+    const quads = new Readable({objectMode: true});
+    let textStream: Readable;
+    let textBuffer: string = '';
+    let initialized: boolean = false;
+    quads._read = async () => {
+      if (!initialized) {
+        initialized = true;
+        let mediaTypeFound: string;
+        let isTextFound: boolean = false;
+        let countScriptTexts: number = 0; // amount of script-texts that have been found for parsing
+        let ended: boolean = false;
+        let closed: boolean = false; // after onclosetag is called, htmlparser can return an empty text in ontext
```

Comment:
The two variable names above are confusing. I suggest to rename to `streamEnded` and `tagClosed`.

Reply:
Done